app/src/main/java/com/cortado/minionify/ModeChoiceActivity.java

```diff
@@ -0,0 +1,78 @@
+package com.cortado.minionify;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.View;
+import android.view.Window;
+import android.view.WindowManager;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.cortado.minionify.mode.AppMode;
+import com.cortado.minionify.mode.exceptions.ModeNotExistsException;
+import com.cortado.minionify.preference.PreferencesSingleton;
+import com.cortado.minionify.R;
+import com.cortado.minionify.preference.exceptions.PreferenceOperationFailure;
+
+public class ModeChoiceActivity extends AppCompatActivity {
+
+    private PreferencesSingleton mPreferences;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mPreferences = PreferencesSingleton.getInstance(getPreferences(Context.MODE_PRIVATE));
+
+        AppMode appMode = getAppMode();
+        if (appMode != null) {
+            redirectActivity(appMode);
+            findViewById(R.id.manager_button).setOnClickListener(null);
```

Comment:
`.setEnabled(false);`

Reply:
Done, but why does it matter?

Reply:
just the correct way of doing things, and no use of nulls